### PR TITLE
Add support for backward compatible API Client behavior

### PR DESCRIPTION
**Pull request by smallinsky**
## What

Add support for backward-compatible behavior in the case where tsh profile was generated by tsh version without CAS fix where trusted clusters CAs were split and moved to `cas` directory. 

This case occurs when a client uses an old tsh client without CAS fix and a news version of teleport plugins that try to load clusters certs from CAS directory in case of creating auth client from the `tsh` users profile.

### CODE REVIEW

**Review by r0mant — APPROVED** [review]

File: api/utils/keypaths/keypaths.go [hunk 1]
```diff
@@ -144,6 +146,14 @@ func CAsDir(baseDir, proxy string) string {
 	return filepath.Join(ProxyKeyDir(baseDir, proxy), casDir)
 }
 
+// TLSCAsPath returns the path to the users's TLS CA's certificates
+// for the given proxy.
+// <baseDir>/keys/<proxy>/certs.pem
+// DELETE IN 10.0. Deprecated
```
Comment (jakule) [inline api/utils/keypaths/keypaths.go:152]: If this needs to be removed in 10.0 shouldn't we just fix it in 9.0? Technically this code will be deprecated the moment when is merged.

File: api/profile/profile.go [hunk 2]
```diff
@@ -336,6 +377,10 @@ func (p *Profile) TLSClusterCASDir() string {
 	return keypaths.CAsDir(p.Dir, p.Name())
 }
 
+func (p *Profile) TLSCAsPath() string {
```
Comment (jakule) [inline api/profile/profile.go:380]: nit: missing Go doc

File: api/profile/profile.go [hunk 3]
```diff
@@ -129,11 +162,19 @@ func (p *Profile) TLSConfig() (*tls.Config, error) {
...
+}
 
-	return &tls.Config{
-		Certificates: []tls.Certificate{cert},
-		RootCAs:      pool,
-	}, nil
+func certPoolFromLegacyCAFile(p *Profile) (*x509.CertPool, error) {
+	caCerts, err := os.ReadFile(p.TLSCAsPath())
+	if err != nil {
+		return nil, trace.Wrap(err)
```
Comment (jakule) [inline api/profile/profile.go:171]: maybe?
```suggestion
		return nil, trace.ConvertSystemError(err)
```

**Review by jakule — APPROVED** [review]

File: api/utils/keypaths/keypaths.go (hunk 1 above)
Comment (smallinsky) [reply api/utils/keypaths/keypaths.go:152]: The issue is not stricte related to any `teleport` `tsh` or `tctl` flow but occurs in external tools that depends on teleport/api/client` package like  [terraform](https://github.com/gravitational/teleport-plugins/blob/master/go.mod#L20) teleport-plugin where client still uses old `tsh` that generates certs.pem dir  but the teleport-plugin uses a news version of `teleport/api/client`  that doesn't read certs from certs.pem. 

If we will be sure that all clients switched to teleport v9 where trusted certs are stored in CAS directory we will be able to drop support for `certs.pem` file.

**Pull request merged**